static/js/filter.js

```diff
@@ -68,10 +68,6 @@ function message_matches_search_term(message, operator, operand) {
         }
         return true; // in:whatever returns true
 
-    case 'near':
-        // this is all handled server side
-        return true;
-
```

Comment:
Can you explain why you're editing `message_matches_search_term` here?  It's not obvious to me why that would be correct.

Reply:
I mention this in the commit message--the code is unreachable, since `near` queries cannot be applied locally and the higher level predicate function short circuits those.  Also, the `default` section for `message_matches_search_term` returns `true`, so that `case` has always been redundant in a sense.  Basically, from filtering, the notion of "near" is undefined, or perhaps more precisely, always `true` bye definition.